Assets/Scripts/Tools/DownloadFile.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.Networking;
+using UnityEngine.UI;
+
+namespace Netherlands3D.Twin
+{
+    public class DownloadFile : MonoBehaviour
+    {
+        [SerializeField] private string[] validFileExtensions;
+        public string URL { get; set; }
+        public UnityEvent<string> onFileDownloaded = new();
+        public UnityEvent<string> onFileDownloadFailed = new();
+
+        private void Awake()
+        {
+            for (var i = 0; i < validFileExtensions.Length; i++)
+            {
+                var extension = validFileExtensions[i].Trim();
+                if (!extension.StartsWith('.'))
+                {
+                    extension = "." + extension;
+                }
+
+                validFileExtensions[i] = extension;
+            }
+        }
+
+        //for in the inspector
+        public void SetURL(string url)
+        {
+            URL = url;
+        }
+
+        public void DownloadFileFromURL()
+        {
+            StartCoroutine(Download(URL));
+        }
+
+        private IEnumerator Download(string url)
+        {
+            string filename = string.Empty;
+            try
+            {
+                Uri uri = new Uri(url);
+                filename = Path.GetFileName(uri.AbsolutePath);
+
+                if (filename == string.Empty)
+                {
+                    Debug.LogError("The provided URL does not contain a valid filename: " + url);
+                    onFileDownloadFailed.Invoke("The provided URL does not contain a valid filename: " + url);
+                    yield break;
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogError(e.Message);
+                onFileDownloadFailed.Invoke(e.Message);
+                yield break;
+            }
```

Comment:
Als we (ooit) exceptions uit willen gaan zetten in de build dan is het niet verstandig om try/catch te gebruiken ,-dan zal daar de WebGL build op crashen. Deze zelfde logic kan ook gewoon in een if/else statement

Bij 3DAmsterdam/3DUtrecht gaf het een enorme performance boost [als exception handling uit staat.](https://docs.unity3d.com/ScriptReference/PlayerSettings.WebGL-exceptionSupport.html)